rotkehlchen/db/history_events.py

```diff
@@ -397,40 +441,18 @@ def get_history_events(
         TODO: To not query all columns with all joins for all cases, we perhaps can
         peek on the entry type of the filter and adjust the SELECT fields accordingly?
         """
-        base_prefix = 'SELECT '
-        base_suffix = f'{HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS}, {ETH_STAKING_EVENT_FIELDS} {ALL_EVENTS_DATA_JOIN}'  # noqa: E501
-        bindings = []
-        type_idx = 1 if group_by_event_ids is True else 0
-        special_free_query = False
-        free_query_group_by = maybe_filter_ignore_asset(filter_query)
-
-        if has_premium is True:
-            if group_by_event_ids is True:
-                base_prefix += 'COUNT(*), '
-            base_query = f'{base_prefix} {base_suffix}'
-        else:
-            if group_by_event_ids is True:
-                special_free_query = True  # a bit ugly conditions to keep limit at groups for free users  # noqa: E501
-                free_query_group_by += 'GROUP BY event_identifier'
-                base_query = (
-                    f'{base_prefix} * FROM (SELECT COUNT(*), {base_suffix} '
-                    f'{free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '
-                )
-            else:  # if we don't group them but still apply free limit, then apply this limit only on any event_identifiers that fall inside the limit  # noqa: E501
-                base_query = (
-                    f'{base_prefix} * FROM (SELECT {base_suffix} WHERE event_identifier IN '
-                    f'(SELECT DISTINCT event_identifier FROM history_events {free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?)) '  # noqa: E501
-                )
-            bindings = [FREE_HISTORY_EVENTS_LIMIT]
+        base_query, filters_bindings = self._create_history_events_query(
+            filter_query=filter_query,
+            group_by_event_ids=group_by_event_ids,
+            entries_limit=FREE_HISTORY_EVENTS_LIMIT,
+        )[0 if has_premium else 1]
```

Comment:
I see how this improved the function but the list indexing here is a bad pattern. `has_premium` should be an argument and in `_create_history_events_query` you should create only the strings that you need. You can have the if has_premium in there and only create and return what you need

Reply:
This is error prone and harder to follow